test/jdk/sun/security/ssl/ALPN/AlpnGreaseTest.java

```diff
@@ -0,0 +1,306 @@
+
+    private final SSLEngine serverEngine;     // server Engine
+    private final ByteBuffer serverOut;       // write side of serverEngine
+    private final ByteBuffer serverIn;        // read side of serverEngine
+
+    // For data transport, this example uses local ByteBuffers.  This
+    // isn't really useful, but the purpose of this example is to show
+    // SSLEngine concepts, not how to do network transport.
+    private final ByteBuffer cTOs;      // "reliable" transport client->server
+    private final ByteBuffer sTOc;      // "reliable" transport server->client
+
+    // These are the various 8-bit char values that could be sent as GREASE
+    // values.  We'll just make one big String here to make it easy to check
+    // that the right values are being output.
+    private static final byte[] greaseBytes = new byte[] {
+        (byte) 0x0A, (byte) 0x1A, (byte) 0x2A, (byte) 0x3A,
+        (byte) 0x4A, (byte) 0x5A, (byte) 0x6A, (byte) 0x7A,
+        (byte) 0x8A, (byte) 0x9A, (byte) 0xAA, (byte) 0xBA,
+        (byte) 0xCA, (byte) 0xDA, (byte) 0xEA, (byte) 0xFA
+    };
+
+    private static final String greaseString =
+            new String(greaseBytes, StandardCharsets.ISO_8859_1);
+
+    private static void findGreaseInClientHello(byte[] bytes) throws Exception {
+        for (int i = 0; i < bytes.length - greaseBytes.length; i++) {
+            if (Arrays.equals(bytes, i, i + greaseBytes.length,
+                    greaseBytes, 0, greaseBytes.length)) {
+                System.out.println("Found greaseBytes in ClientHello at: " + i);
+                return;
+            }
+        }
+        throw new Exception("Couldn't find greaseBytes");
+    }
+
+    private AlpnGreaseTest() throws Exception {
+        serverEngine = configureServerEngine(
+                createServerSSLContext().createSSLEngine());
+
+        clientEngine = configureClientEngine(
+                createClientSSLContext().createSSLEngine());
+
+        // We'll assume the buffer sizes are the same
+        // between client and server.
+        SSLSession session = clientEngine.getSession();
+        int appBufferMax = session.getApplicationBufferSize();
+        int netBufferMax = session.getPacketBufferSize();
+
+        // We'll make the input buffers a bit bigger than the max needed
+        // size, so that unwrap()s following a successful data transfer
+        // won't generate BUFFER_OVERFLOWS.
+        //
+        // We'll use a mix of direct and indirect ByteBuffers for
+        // tutorial purposes only.  In reality, only use direct
+        // ByteBuffers when they give a clear performance enhancement.
+        clientIn = ByteBuffer.allocate(appBufferMax + 50);
+        serverIn = ByteBuffer.allocate(appBufferMax + 50);
+
+        cTOs = ByteBuffer.allocateDirect(netBufferMax);
+        sTOc = ByteBuffer.allocateDirect(netBufferMax);
+
+        clientOut = ByteBuffer.wrap("Hi Server, I'm Client".getBytes());
```

Comment:
`.getBytes()` uses the platform's default charset. Is this intentional ?

Reply:
I am not able to see the comment here in github due to the Terms of Use issue flagged by github, but did get email with the issue so I'm hoping/assuming that the terms have been accepted.

For this test, I just took the standard JSSE test template and added the few lines to check for the proper ALPN exchange.

I'm 99% sure that for the JDK testsuite, the default Charset is UTF-8 which will accept ASCII values like this.  If not, there's going to be a lot more problems.  :)